File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    
    dict = {}
    
    for idx, alpha in enumerate(s):
        if alpha not in dict:
            answer.append(-1)
            dict[alpha] = idx
        else:
            answer.append(idx - dict[alpha])
            dict[alpha] = idx
    
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    
    let strMap = new Map();
    for (let i = 0; i < s.length; i++) {
        if (strMap.get(s[i]) === undefined) {
            answer.push(-1);
            strMap.set(s[i], i);
        } else {
            answer.push(i - strMap.get(s[i]));
            strMap.set(s[i], i);
        }
    }
    
    return answer;
}